commit with changes for wsdl2rest generator

bfitzpat opened the pull request:
This is NOT TO BE MERGED at present, but includes a second attempt of making the wsdl2rest generator a sub-generator of the camel-project generator.

In the ./generators/wsdl2rest directory, run:

> mvn clean install

This will pull the needed Java files into the ./generators/wsdl2rest/target directory.

We currently have an issue with running the Java from within the uber-jar. To run the executable jar, simply run the testme.sh or testme.bat script to call the Java class with the right settings. At this time, we get:

```
Error executing command
org.apache.cxf.tools.common.ToolException: Could not find jaxws frontend within classpath
        at org.apache.cxf.tools.wsdlto.core.PluginLoader.getFrontEnd(PluginLoader.java:254)
        at org.apache.cxf.tools.wsdlto.core.PluginLoader.getFrontEndProfile(PluginLoader.java:389)
        at org.apache.cxf.tools.wsdlto.WSDLToJava.loadFrontEnd(WSDLToJava.java:64)
        at org.apache.cxf.tools.wsdlto.WSDLToJava.run(WSDLToJava.java:96)
        at org.apache.cxf.tools.wsdlto.WSDLToJava.run(WSDLToJava.java:86)
        at org.jboss.fuse.wsdl2rest.impl.codegen.JavaTypeGenerator.execute(JavaTypeGenerator.java:36)
        at org.jboss.fuse.wsdl2rest.impl.Wsdl2Rest.process(Wsdl2Rest.java:77)
        at org.jboss.fuse.wsdl2rest.impl.Main.mainInternal(Main.java:60)
        at org.jboss.fuse.wsdl2rest.impl.Main.main(Main.java:36)

```

ffang commented:
Hi Fitz,

For the uber-jar, I believe you also need to combine all files under META-INF/ from different jars, such as META-INF/cxf/bus-extensions.txt and META-INF/spring.handlers, otherwise you will run into issues.

You can find more details from here[1]  
and how other project like tomee generate uber jar for CXF[2]

[1]http://cxf.apache.org/docs/bundling-cxf-into-single-jar-with-maven-shade-plugin.html
[2]https://tomee.apache.org/advanced/shading/index.pdf

Cheers
Freeman

bfitzpat commented:
@lhein I have gone back to a non-uber-jar approach with this latest commit and rebased with your last changes to master. If you do a "mvn clean install" in the wsdl2rest directory and run the testme2 script, you should see something to the effect of: 
'''
Exception in thread "main" java.lang.NoClassDefFoundError: org/slf4j/LoggerFactory
        at org.jboss.fuse.wsdl2rest.impl.Main.<clinit>(Main.java:32)
Caused by: java.lang.ClassNotFoundException: org.slf4j.LoggerFactory
        at java.net.URLClassLoader.findClass(Unknown Source)
        at java.lang.ClassLoader.loadClass(Unknown Source)
        at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
        at java.lang.ClassLoader.loadClass(Unknown Source)
        ... 1 more
'''

That's what I bashed my head against yesterday all day.

bfitzpat commented:
But to answer your question from Mattermost: 
* uber-jar request originally came from Aurelien 
* originally I had it working locally in a manner similar to what I just committed, but now we're getting the weird logging issue I can't figure out

bfitzpat commented:
@lhein I have updated the pom.xml to simply grab the wsdl2rest-dist distribution, unpack it, and copy over the libs. For some reason the lib directory isn't going away during the clean, but I have also updated the scripts (see testme3.sh/testme3.bat) and we still get the logging error.

I also updated the mocha tests to they run in your CI.

bfitzpat commented:
@lhein Ok, so my last attempt for today. I found an issue with the camel version we were pulling in. It didn't jive with the version of wsdl2rest. Now I have that working and am copying the dependent files from the distribution into the lib directory. That part works.

If you then run "./bin/wsdl2rest.sh" in the wsdl2rest directory, the shell script works just like the distribution does and you get an expected result. 

Unfortunately, I am unable to simply call the java via the jar (I get a logger not found issue) and if I call the script directly I get a similar result:
```
PS C:\Users\brianf\Documents\GitHub\generator-camel-project-fork\testproj> yo camel-project:wsdl2rest

Just found a `.yo-rc.json` in a parent directory.
Setting the project root at: C:\Users\brianf\Documents\GitHub\generator-camel-project-fork
? URL to the input WSDL file:/E:/eclipse-photon/eclipse-workspace/empty-spring/src/main/resources/META-INF/wsdl/Address.wsdl
? Name of the output directory for generated artifacts src//main//java
wsdl url file:/E:/eclipse-photon/eclipse-workspace/empty-spring/src/main/resources/META-INF/wsdl/Address.wsdl
output path src//main//java
wsdl url file:/E:/eclipse-photon/eclipse-workspace/empty-spring/src/main/resources/META-INF/wsdl/Address.wsdl
calling: sh C:\Users\brianf\Documents\GitHub\generator-camel-project-fork\generators\wsdl2rest\bin\wsdl2rest.sh --wsdl file:/E:/eclipse-photon/eclipse-workspace/empty-spring/src/main/resources/META-INF/wsdl/Address.wsdl --out C:\Users\brianf\Documents\GitHub\generator-camel-project-fork\src\main\java
stderr: log4j:ERROR Could not read configuration file from URL [file:/c/Users/brianf/Documents/GitHub/generator-camel-project-fork/generators/wsdl2rest/config/logging.properties].
stderr:

stderr: java.io.FileNotFoundException: \c\Users\brianf\Documents\GitHub\generator-camel-project-fork\generators\wsdl2rest\config\logging.properties (The system cannot find the path specified)

stderr:         at java.io.FileInputStream.open0(Native Method)
        at java.io.FileInputStream.open(Unknown Source)
        at java.io.FileInputStream.<init>(Unknown Source)
        at java.io.FileInputStream.<init>(Unknown Source)
        at sun.net.www.protocol.file.FileURLConnection.connect(Unknown Source)
        at sun.net.www.protocol.file.FileURLConnection.getInputStream(Unknown Source)
        at org.apache.log4j.PropertyConfigurator.doConfigure(PropertyConfigurator.java:557)
        at org.apache.log4j.helpers.OptionConverter.selectAndConfigure(OptionConverter.java:526)
        at org.apache.log4j.LogManager.<clinit>(LogManager.java:127)
        at org.slf4j.impl.Log4jLoggerFactory.<init>(Log4jLoggerFactory.java:66)
        at org.slf4j.impl.StaticLoggerBinder.<init>(StaticLoggerBinder.java:72)
        at org.slf4j.impl.StaticLoggerBinder.<clinit>(StaticLoggerBinder.java:45)
        at org.slf4j.LoggerFactory.bind(LoggerFactory.java:156)
        at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:130)
        at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:424)
        at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:365)
        at org.slf4j.LoggerFactory.getLogger(LoggerFactory.java:391)

stderr:         at org.jboss.fuse.wsdl2rest.impl.Main.<clinit>(Main.java:32)
log4j:ERROR Ignoring configuration file [file:/c/Users/brianf/Documents/GitHub/generator-camel-project-fork/generators/wsdl2rest/config/logging.properties].ogging.properties].

stderr: log4j:WARN No appenders could be found for logger (org.jboss.fuse.wsdl2rest.impl.Main).
stderr:
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.

child process exited with code 1
```

bfitzpat commented:
Some progress. At least I can run the utility in the directory where we're pulling the files in. I am still stuck about getting it to run from inside the generator however and unsure as to the issues...

lhein commented:
One thing I experienced is that I am again lacking a repository to download the wsdl2rest distributions. Seems you still stick to your settings.xml instead of using repository definitions inside the pom. Anyways that used distro is EA and therefor not available outside vpn anyways.

bfitzpat commented on generators/wsdl2rest/pom.xml:
```diff
@@ -0,0 +1,325 @@
+        </plugins>
+    </build>
+
+    <repositories>
```
@lhein The repositories are right here

bfitzpat commented:
With the latest changes, you can now run "yo camel-project:wsdl2rest" and if you provide a good file URL to a wsdl file and a path, it will generate code and config in the location you specify. Huzzah!

apupier commented on generators/wsdl2rest/config/logging.properties:
```diff
@@ -0,0 +1,38 @@
+###
+# #%L
+# Fuse Patch :: Distro :: Standalone
```
Fuse Patch Distro?

apupier commented on generators/wsdl2rest/config/logging.properties:
```diff
@@ -0,0 +1,38 @@
+# #%L
+# Fuse Patch :: Distro :: Standalone
+# %%
+# Copyright (C) 2015 Private
```
2015?

apupier commented on generators/wsdl2rest/pom.xml:
```diff
@@ -0,0 +1,325 @@
+                                <!-- Impl -->
+                                <artifactItem>
+                                    <groupId>org.jboss.fuse.wsdl2rest</groupId>
+                                    <artifactId>wsdl2rest-dist</artifactId>
```
wsd2rest-dist is not provided on public repository, see https://issues.jboss.org/browse/ENTESB-9243

bfitzpat commented:
Spun off a new PR that is up to date with master. https://github.com/camel-tooling/generator-camel-project/pull/31
